VenafiPS/Public/Add-TppAdaptableHash.ps1

```diff
@@ -0,0 +1,179 @@
+
+    .LINK
+    https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-add.php
+
+    .LINK
+    https://docs.venafi.com/Docs/currentSDK/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-ownerdelete.php
+
+    .LINK
+    https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-retrieve.php
+    #>
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( {
+                if ( $_ | Test-TppDnPath ) {
+                    $true
+                } else {
+                    throw "'$_' is not a valid DN path"
+                }
+            })]
+        [Alias('DN')]
+        [String] $Path,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [switch] $AdaptableApp,
+
+        [Parameter()]
+		[string] $Keyname = "Software:Default",
+
+        [Parameter(Mandatory)]
+        [Alias('File')]
+        [string] $FilePath,
+
+        [Parameter()]
+        [psobject] $VenafiSession = $VenafiSession
+    )
+
+    begin {
+        Test-VenafiSession -VenafiSession $VenafiSession -Platform 'TPP'
+
+        $params = @{
+            VenafiSession = $VenafiSession
+            Method        = 'Post'
+        }
+
+        if ( $AdaptableApp ) {
+            $retrieveVaultID = ( Get-TppAttribute -Path $Path -Class 'Adaptable App' -Attribute 'PowerShell Script Hash Vault Id' ).'PowerShell Script Hash Vault Id'
+        } else {
+            $retrieveVaultID = ( Get-TppAttribute -Path $Path -Attribute 'PowerShell Script Hash Vault Id' ).'PowerShell Script Hash Vault Id'
+        }
+
+        $bytes = [Text.Encoding]::UTF32.GetBytes([IO.File]::ReadAllText($FilePath))
+        $hash = Get-FileHash -InputStream ([System.IO.MemoryStream]::New($bytes))
+        $base64data = [Convert]::ToBase64String([Text.Encoding]::UTF8.GetBytes($hash.hash.ToLower()))
+
+    }
+
+    process {
+        if ( -not $PSCmdlet.ShouldProcess($Path) ) {
+            continue
+        }
+
+        if ( $retrieveVaultID ) {
+            $paramsretrieve = $params.Clone()
+            $paramsretrieve.UriLeaf = 'SecretStore/retrieve'
+            $paramsretrieve.Body = @{
+                VaultID = $retrieveVaultID
+            }
+
+            $retrieveResponse = Invoke-VenafiRestMethod @paramsretrieve
+
+            if ( $retrieveResponse.Result -ne [TppSecretStoreResult]::Success ) {
+                Write-Error ("Error retrieving VaultID: {0}" -f [enum]::GetName([TppSecretStoreResult], $retrieveResponse.Result)) -ErrorAction Stop
+            }
+
+            if($null -ne $retrieveResponse.Base64Data) {
+                $retrieveBase64 = $retrieveResponse.Base64Data
+            }
+        }
+
+        if ( $base64data -eq $retrieveBase64 ){
+            Write-Verbose "PowerShell Script Hash Vault Id unchanged for $($Path)."
+            continue
+        } else {
```

Comment:
this works fine and if your preference definitely leave as is, but I try not to have huge if/else as they get difficult to read.  given the continue, you could end the if and remove the else

Reply:
that if statement isn't the last bit of code in the function.  if it was I'd agree with you.  since it's not, there's always a chance something will be added further down the line and get executed mistakenly.